jcl/src/jdk.management/share/classes/openj9/lang/management/internal/OpenJ9DiagnosticsMXBeanImpl.java

```diff
@@ -0,0 +1,379 @@
+/*[INCLUDE-IF Sidecar17]*/
+/*******************************************************************************
+ * Copyright (c) 2018, 2018 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+
+package openj9.lang.management.internal;
+
+/*[IF Sidecar19-SE]*/
+/*[IF Sidecar19-SE-OpenJ9]*/
+import java.lang.Module;
+import java.lang.ModuleLayer;
+/*[ELSE]
+import java.lang.reflect.Module;
+import java.lang.reflect.Layer;
+/*[ENDIF]*/
+/*[ENDIF]*/
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import javax.management.MalformedObjectNameException;
+import javax.management.ObjectName;
+/*[IF !Sidecar19-SE]
+import com.ibm.jvm.Dump;
+import com.ibm.jvm.InvalidDumpOptionException;
+import com.ibm.jvm.DumpConfigurationUnavailableException;
+/*[ENDIF]*/
+
+import openj9.lang.management.OpenJ9DiagnosticsMXBean;
+import openj9.lang.management.InvalidOptionException;
+import openj9.lang.management.ConfigurationUnavailableException;
+
+/**
+ * Runtime type for {@link OpenJ9DiagnosticsMXBean}.
+ * <p>
+ * Implements functionality to dynamically configure dump options and to trigger supported dump agents.
+ * </p>
+ */
+public final class OpenJ9DiagnosticsMXBeanImpl implements OpenJ9DiagnosticsMXBean {
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void resetDumpOptions() throws ConfigurationUnavailableException {
+		try {
```

Comment:
Although the dump APIs already check security permissions, it may be useful to have an extra layer of permission checking for the MXBean, since the MXBean can be used remotely. Please add a security check for java.lang.ManagementPermission("control") (i.e. ManagementPermissionHelper.MPCONTROL) to each of the APIs. This allows the local application to have DumpPermssion, but if ManagementPermission("control") is not granted then the MXBean APIs cannot be used.

Reply:
ManagementPermissionHelper is an internal class in java.management module, not sure if there would be any restrictions in accessing internal classes using reflection from another module in future. Instead can I just use ManagementPermission("control") directly?

Reply:
Yes, you can use `ManagementPermission("control")` directly, but you should capture that object somewhere (like ManagementPermissionHelper does) so that you can reuse it.

Reply:
java.management already `exports com.ibm.java.lang.management.internal to jdk.management;` so I think ManagementPermissionHelper can be used directly.